packages/testkit-backend/src/infrastructure/http-static.server.js

```diff
@@ -0,0 +1,66 @@
+import fs from 'fs'
+
+const DEFAULT_MIME_TYPES = {
+  html: 'text/html',
+  js: 'text/javascript'
+}
+
+const DEFAULT_MIME_TYPE = 'text/plain'
+
+const DEFAULT_INDEX_FILE = '/index.html'
+
+/**
+ * This a very minimal implementation of a static files http server.
+ *
+ * This is intend to be used only in the ${@link RemoteController} scenario and
+ * this is not feature complete.
+ *
+ * Some security measures to avoid external elements to sniff to files out of the base
+ * path are in place. Like Directory Transversal.
+ */
+export class HttpStaticServer {
+  constructor ({
+    basePath,
+    mimeTypes,
+    defaultMimeType,
+    indexFile
+  } = {}) {
+    this._basePath = basePath
+    this._mimeTypes = mimeTypes || DEFAULT_MIME_TYPES
+    this._indexFile = indexFile || DEFAULT_INDEX_FILE
+    this._defaultMimeType = defaultMimeType || DEFAULT_MIME_TYPE
+  }
+
+  serve (request, response) {
+    if (request.method === 'GET') {
+      const file = request.url !== '/' ? request.url : this._indexFile
+      const filePath = `${this._basePath}${file}`
+      const [, extension, ...rest] = file.split('.')
+      if (rest.length > 0) {
```

Comment:
A bit simplistic :D doesn't allow valid usage of `..` (staying withing `basePath`) nor files or folders that make the path contain multiple `.` (e.g., `my_file.txt.bak`). But should be good enough for TestKit if the limitations are well understood.